tests/unit/utils/kubeapi/ingress.go

```diff
@@ -22,25 +22,30 @@ func FindIngressRuleByHttpPath(rules []networkingv1.IngressRule, path string) (*
 // VerifyNoRuleWithPath asserts that none of the given IngressRules have a given HTTP path entry.
 func VerifyNoRuleWithPath(rules []networkingv1.IngressRule, path string) {
 	authRule, _ := FindIngressRuleByHttpPath(rules, path)
-	ExpectWithOffset(1, authRule).To(BeNil())
```

Comment:
How come you changed from `ExpectWithOffset` to `Expect`? The former reports the line number at which `VerifyNoRuleWithPath` is called which I think is pretty helpful.

Reply:
During debugging, I found that `...WithOffset(1,...` in this specific case obscured useful information. The line number you're referring to is also in the output and easily identified.

Reply:
>The line number you're referring to is also in the output and easily identified.

Maybe in this particular case but not always. For example, it won't be easily identifiable if I call the verify function multiple times in the test. 

To illustrate I modified the `VerifyNoRuleWithPath` function to fail. Here is what is the output looks like:

```
• Failure [0.317 seconds]
Verify Stargate ingress template
/Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/template_stargate_ingress_test.go:14
  by rendering it with options
  /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/template_stargate_ingress_test.go:95
    using only default options [It]
    /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/template_stargate_ingress_test.go:96

    Expected
        <*v1beta1.IngressRule | 0x0>: nil
    not to be nil

    /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/utils/kubeapi/ingress.go:26
```
Line 96 is the declaration of the test spec. It does not tell me the exact line where the failure is. 

Next I modified `VerifyNoRuleWithPath` to fail and use `ExpectWithOffset(1, ...)`. Here is the output:

```
• Failure [0.313 seconds]
Verify Stargate ingress template
/Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/template_stargate_ingress_test.go:14
  by rendering it with options
  /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/template_stargate_ingress_test.go:95
    using only default options [It]
    /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/template_stargate_ingress_test.go:96

    Expected
        <*v1beta1.IngressRule | 0x0>: nil
    not to be nil

    /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/template_stargate_ingress_test.go:178
```
Line 178 points to the call made from within `verifyIngressRules`.

But if I change it to use `ExpectWithOffset(2, ...)` my output is what I want:

```
• Failure [0.323 seconds]
Verify Stargate ingress template
/Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/template_stargate_ingress_test.go:14
  by rendering it with options
  /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/template_stargate_ingress_test.go:95
    using only default options [It]
    /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/template_stargate_ingress_test.go:96

    Expected
        <*v1beta1.IngressRule | 0x0>: nil
    not to be nil

    /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/template_stargate_ingress_test.go:107
```
Now I know precisely at what line in the spec where the failure occurred. I think this is better but it certainly isn't perfect. We could refactor the the verify function to take an offset argument but as code gets refactored that offset could easily wind up pointing to the wrong place in the call stack.

I think the best solution and probably most idiomatic Go solution is to have the verify functions return an `error`. This will make it trivial to know where in the test spec the failure happens and good, descriptive error message should let us know what exactly failed.

Reply:
@jakerobb Let's leave the functions as is. I don't want to hold up the PR on this. I will create a follow up ticket where we can look at this more closely and decide what to do. It can be done post-merge.

Reply:
Okay. Note that, in later revisions, I added a mismatch description to the Expect calls. This provides sufficient context to identify the line in the calling function.